test/libyul/objectCompiler/eof/creation_with_immutables.yul

```diff
@@ -0,0 +1,100 @@
+object "a" {
+    code {
+        mstore(0, eofcreate("b", 0, 0, 0, 0))
+        return(0, 32)
+    }
+
+    object "b" {
+        code {
+            mstore(0, 0x1122334455667788990011223344556677889900112233445566778899001122)
+            mstore(32, 0x1122334455667788990011223344556677889900112233445566778899001122)
+            returncontract("c", 0, 64)
+        }
+        object "c" {
+            code {
+                let d0 := auxdataloadn(0)
+                let d1 := auxdataloadn(32)
+
+                mstore(0, d0)
+                mstore(32, d1)
+
+                return(0, 64)
+            }
+        }
+   }
+
+    data "data1" hex"48656c6c6f2c20576f726c6421"
+}
+
+// ====
+// bytecodeFormat: >=EOFv1
+// EVMVersion: >=prague
+// ----
+// Assembly:
+//     /* "source":80:81   */
+//   0x00
+//     /* "source":56:82   */
+//   dup1
+//   dup1
+//   dup1
+//   eofcreate(0)
```

Comment:
This actually looks a bit ambiguous, as if `EOFCREATE` was getting only one argument.

How about we change the notation to for immediates in asm output to `{}` in `AssemblyItem::toAssemblyText)`? Then we'd get this:

```suggestion
//   eofcreate{0}
```

I'd do that for `RETURNCONTRACT` and `AUXDATALOADN` as well. And I'd also return true for `RETURNCONTRACT` from `AssemblyItem::canBeFunctional()`. Is there actually a good reason why it's not like that already?

Reply:
I will change to brackets in `AssemblyItem::toAssemblyText`. Fot DATALOADN I would do this in a separated PR.

Regarding `ssemblyItem::canBeFunctional()`
It depends how you define functional context. I assumed that if there is no return value it's not functional.

Reply:
> It depends how you define functional context. I assumed that if there is no return value it's not functional.

`canBeFunctional()` determines whether the parameters of the item should be printed in parentheses after it (in functional style) or as separate instructions before it.

At least that's how I understand it from how `Functionalizer` uses it: https://github.com/ethereum/solidity/blob/879d8e69f884419a9bc4a0cfe17a6d73ae14d836/libevmasm/Assembly.cpp#L324-L345

Here `m_pending` is a list of previous expressions, which are assumed to be the arguments.

I think that we want `true` for almost everything. The only things for which we return false seem to be `PUSH` and `DUP` instructions and `Tag`, `AssignImmutable`, `VerbatimBytecode` items. Not sure about the reasons for all of them but e.g. `DUP` is defined as taking the whole top of the stack as arguments (so e.g. `DUP16` takes 17 arguments).

The closest thing we have to `RETURNCONTRACT` is probably `RETURN` and for that we return `true`.

Reply:
BTW, I wonder if it wouldn't be better to show sub names like we do for `dataOffset`/`dataSize` instead of the numeric container IDs.

So for example in the assembly we would print `eofcreate{sub_0}` rather than `eofcreate{0}`.

Though I guess with EOF the ID approach is actually viable because we can't have nesting so maybe the current approach is still good. It's something to consider though.

Reply:
First fixed. I will check the later.